components/Map.js

```diff
@@ -0,0 +1,61 @@
+import "mapbox-gl/dist/mapbox-gl.css";
+import mapboxgl from "!mapbox-gl";
+import getCoordinates from "@/lib/getCoordinates";
+import { useRef, useState, useEffect } from "react";
+import { set } from "mongoose";
+
+mapboxgl.accessToken = process.env.NEXT_PUBLIC_MAPBOX_TOKEN;
+
+export default function Map({ eventAddress, setCoordinates }) {
```

Comment:
we are not drilling setter functions, its bad practice. Check the [react docu ](https://react.dev/reference/react/useState#counter-number)